pkg/apis/pravega/v1beta1/pravegacluster_types.go

```diff
@@ -1065,8 +1069,10 @@ func (p *PravegaCluster) validateConfigMap() error {
 	}
 	if val, ok := p.Spec.Pravega.Options["bookkeeper.ledger.path"]; ok {
 		checkstring := fmt.Sprintf("-Dbookkeeper.ledger.path=%v", val)
-		eq := strings.Contains(configmap.Data["JAVA_OPTS"], checkstring)
-		if !eq {
+		eq_new := strings.Contains(configmap.Data["JAVA_OPTS"], checkstring)
```

Comment:
In place of contains let's check if the configmap.Data["JAVA_OPTS"] is equal to the checkstring or not to avoid any false positives.
Please change this at all other places too.

Reply:
We cannot use the equals check here since this particular option is just one of the options within the list of java options for pravega.

Reply:
what I meant is let's think of a logic where we should check if specific option field value is equal to the changed value or not.
for eg think of a case where a user tries to change ledger path from **ledgers/temp1 to ledgers**   this will pass our check even though the ledger path is changed.